arch/arm/core/cortex_r/Kconfig

```diff
@@ -0,0 +1,98 @@
+# Kconfig - ARM Cortex-R platform configuration options
+
+#
+# Copyright (c) 2018 Marvell
+# Copyright (c) 2018 Lexmark International, Inc.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+# NOTE: We have the specific core implementations first and outside of the
+# if CPU_CORTEX_R block so that SoCs can select which core they are using
+# without having to select all the options related to that core.  Everything
+# else is captured inside the if CPU_CORTEX_R block so they are not exposed
+# if one selects a different ARM Cortex Family (Cortex-A or Cortex-M)
+
+
+config CPU_CORTEX_R4
+	bool
+	select CPU_CORTEX_R
+	select ARMV7_R
+	select ARMV7_M_ARMV8_M_FP if CPU_HAS_FPU
+	help
+	  This option signifies the use of a Cortex-R4 CPU
+
+if CPU_CORTEX_R
+
+config ARMV7_R
+	bool
+	select ATOMIC_OPERATIONS_BUILTIN
+	select ISA_ARM
+	help
+	  This option signifies the use of an ARMv7-R processor
+	  implementation, or the use of an ARMv8-R processor
+	  implementation supporting the Main Extension.
```

Comment:
Please, clarify: does the -R- architecture have Base and Mainline versions for v8? If not, the help text needs update

Reply:
Btw, I searched a bit and I didn't see clear indication of Baseline, Mainline terminology outside the -M- architecture...